src/modules/ibc.ts

```diff
@@ -155,14 +164,116 @@ export class IBCModule extends BaseModule {
     return chainInfoMap;
   }
 
-  async getChainInfo(chainName: string, chainId: string): Promise<ChainInfo | undefined> {
+  async getChainInfo(chainName: string): Promise<ChainInfo | undefined> {
     if (!chainName) return undefined
-  
-    try {
-      const chainInfoResponse = await fetch(`https://raw.githubusercontent.com/chainapsis/keplr-chain-registry/master/cosmos/${chainName}.json`);
-      const chainInfoJson = await chainInfoResponse.json();
-      return chainInfoJson as ChainInfo;
-    } catch (err) {
+    const chainInfoResponse = await fetch(`https://raw.githubusercontent.com/chainapsis/keplr-chain-registry/master/cosmos/${chainName}.json`)
+    if (!chainInfoResponse.ok) {
+      if (chainInfoResponse.status === 404){
+        return undefined;
+      } 
+    }
+    const chainInfoJson = await chainInfoResponse.json();
+    return chainInfoJson as ChainInfo;
+  }
+
+  async getAssembledChainInfo(chainsData: ChainRegistryItem[], chainId: string, bridgeChainName: string): Promise<ChainInfo | undefined> {
+    const selectedChainData = chainsData.find((chainData: ChainRegistryItem) => {
+      return chainData.chain_name.includes(bridgeChainName.toLowerCase()) || chainData.chain_id.includes(bridgeChainName.toLowerCase());
+    });
+
+    if (selectedChainData) {
+      try {
+        const chainInfoResponse = await fetch(`https://raw.githubusercontent.com/cosmos/chain-registry/master/${selectedChainData.chain_name}/chain.json`);
+        const chainInfoJson = await chainInfoResponse.json();
+        const chainAssetListResponse = await fetch(`https://raw.githubusercontent.com/cosmos/chain-registry/master/${selectedChainData.chain_name}/assetlist.json`);
+        const assetListJson = await chainAssetListResponse.json();
+        const features: string[] = [];
+      if (selectedChainData.cosmwasm_enabled) {
+        features.push("cosmwasm");
+      }
+
+      // Extract the staking currency denomination from chainInfoJson (min. denom)
+      const stakingCurrencyDenom = chainInfoJson['staking']['staking_tokens'][0]['denom'];
+      
+      // Find the asset that matches the stakingCurrencyDenom
+      const stakingAsset = assetListJson.assets.find((asset: Asset) => {
+        return asset.denom_units.some((unit: any) => unit.denom === stakingCurrencyDenom);
+      });
+
+      // Find the highest denom unit
+      const maxExponentDenomUnit = stakingAsset.denom_units.reduce((prev: DenomUnit, current: DenomUnit) => {
+        return (prev.exponent > current.exponent) ? prev : current;
+      });
+
+      let stakeCurrency: Currency = {
+        coinDenom: maxExponentDenomUnit.denom.toUpperCase(),
+        coinMinimalDenom: stakingCurrencyDenom,
+        coinDecimals: maxExponentDenomUnit.exponent,
+        coinGeckoId: stakingAsset.coingecko_id,
```

Comment:
For stakingAsset, make sure to use optional chaining + provide a fallback value as Array.find() function is very likely to return null value.
```typescript
let stakeCurrency: Currency = {
  ...,
  coinGeckoId: stakingAsset?.coingecko_id ?? '',
}
```